src/TMazes.jl

```diff
@@ -219,7 +216,7 @@ function stateindex(maze::TMaze, s::TMazeState)
     end
 end
 
-function generate_o(maze::TMaze, s::TMazeState, rng::AbstractRNG)
+function gen(::DDNOut{:o}, maze::TMaze, s::TMazeState, rng::AbstractRNG)
```

Comment:
It seems like the observation only depends on sp, so we should just implement `observation(::TMaze, sp)` and no other methods of `observation` or `gen`

Reply:
Would this `observation` return a `BoolDistribution`?